spy.go

```diff
@@ -42,3 +103,65 @@ func Execute(w io.Writer, commandLine string) error {
 
 	return nil
 }
+
+// NewSession creates and returns a new shell session with a file
+func NewSession(filepath string, opts ...Option) (session, error) {
+	s := session{}
+	f, err := os.Create(filepath)
+	if err != nil {
+		return session{}, err
+	}
+
+	s.Input = os.Stdin
+	s.Terminal = os.Stdout
+	s.Recorder = f
+
+	for _, opt := range opts {
+		err := opt(&s)
+		if err != nil {
+			return session{}, err
+		}
+	}
+	return s, nil
+}
+
+func (s *session) Run() error {
+	scanner := bufio.NewScanner(s.Input)
+
+	user, err := user.Current()
```

Comment:
Getting the user and hostname feels like it belongs in some helper function (and we should probably call it from the constructor, rather than from here, shouldn't we?)